src/bundles/arcade_2d/functions.ts

```diff
@@ -0,0 +1,891 @@
+/**
+ * The module `arcade_2d` is a wrapper for the Phaser.io game engine.
+ * It provides functions for manipulating GameObjects in a canvas.
+ *
+ * A *GameObject* is defined by its transform and rendered object.
+ *
+ * @module arcade_2d
+ * @author Titus Chew Xuan Jun
+ * @author Xenos Fiorenzo Anong
+ */
+
+// import { context } from 'js-slang/moduleHelpers';
+import Phaser from 'phaser';
+import {
+  PhaserScene,
+  inputKeysDown,
+  pointerPosition,
+  pointerPrimaryDown,
+  pointerOverGameObjectsId,
+  pointerSecondaryDown,
+  debugLogArray,
+} from './phaserScene';
+
+import {
+  type GameObject, RenderableGameObject,
+  type ShapeGameObject,
+  SpriteGameObject,
+  TextGameObject,
+  RectangleGameObject,
+  CircleGameObject,
+  TriangleGameObject,
+  type InteractableGameObject,
+} from './gameobject';
+
+import {
+  type DisplayText,
+  type InteractableProps,
+  type RenderProps,
+  type BuildGame,
+  type TransformProps,
+  type Sprite,
+  type UpdateFunction,
+  type Color,
+  type RectangleProps,
+  type CircleProps,
+  type TriangleProps,
+} from './types';
+
+import {
+  DEFAULT_WIDTH,
+  DEFAULT_HEIGHT,
+  DEFAULT_SCALE,
+  DEFAULT_FPS,
+  MAX_HEIGHT,
+  MIN_HEIGHT,
+  MAX_WIDTH,
+  MIN_WIDTH,
+  MAX_SCALE,
+  MIN_SCALE,
+  MAX_FPS,
+  MIN_FPS,
+  MAX_VOLUME,
+  MIN_VOLUME,
+} from './constants';
+import { AudioClip } from './audio';
+
+// =============================================================================
+// Global Variables
+// =============================================================================
+
+let WIDTH: number = DEFAULT_WIDTH;
+let HEIGHT: number = DEFAULT_HEIGHT;
+let SCALE: number = DEFAULT_SCALE;
+let FPS: number = DEFAULT_FPS;
+export let DEBUG: boolean = false;

```

Comment:
Minor nitpick: perhaps, instead of leaving a bunch of "naked" global variables, it would be better to encapsulate this information in some sort of settings object.

Reply:
Ok, thanks for the suggestion. I have encapsulated the information in a config object.